driven/storage/adapter.go

```diff
@@ -381,22 +381,23 @@ func (m *database) onDataChanged(changeDoc map[string]interface{}) {
 }
 
 // Get Survey
-func (sa *Adapter) GetSurvey(id string) (*model.Survey, error) {
+func (sa *Adapter) GetSurvey(user *model.User, id string) (*model.Survey, error) {
 
 	filter := bson.M{"_id": id}
-	var list []model.Survey
-	err := sa.db.surveys.Find(filter, &list, &options.FindOptions{})
+	var entry model.Survey
+	err := sa.db.surveys.FindOne(filter, &entry, nil)
 	if err != nil {
 		fmt.Printf("error storage.Adapter.GetSurvey(%s) - %s", id, err)
 		return nil, fmt.Errorf("error storage.Adapter.GetSurvey(%s) - %s", id, err)
 	}
 
-	if len(list) > 0 {
-		entry := list[0]
-		return &entry, nil
+	if entry.AppID != user.Claims.AppID || entry.OrgID != user.Claims.OrgID {
+		fmt.Printf("error storage.Adapter.GetSurvey(%s) - 403 incorrect AppID or OrgID", id)
+		return nil, fmt.Errorf("error storage.Adapter.GetSurvey(%s) - 403 incorrect AppID or OrgID", id)
 	} else {
-		return nil, nil
+		return &entry, nil
 	}
```

Comment:
The linter will complain about this. Go style says that you should not use an `else` after a return since it adds unnecessary verbosity. To see these errors, try running the `make` command (specifically `make lint`, but I typically like to run the whole thing just to make sure it builds properly).